xmtp/src/builder.rs

```diff
@@ -48,28 +58,35 @@ where
                 let data_string = std::str::from_utf8(&data).map_err(|e| format!("{}", e))?;
                 let account: VmacAccount =
                     serde_json::from_str(data_string).map_err(|e| format!("{}", e))?;
-                self.account = Some(account)
+                Ok(account)
```

Comment:
In a world with Wallet<>Account associations, the deserialized VmacAccount should be checked against the set wallet address, otherwise it would request a signature (or have some intermediate state like "unlinked")?

Took me a second to grok the reasoning behind setting a wallet address, but it makes sense that we wouldn't set a wallet. We just need to know what wallet this VmacAccount is for or will be for.

Reply:
We also just need to store it somewhere consistent so we can look it up later and re-use on future client instantiations. Wallet address is a good, stable, identifier.